pkg/registry/core/service/storage/rest.go

```diff
@@ -200,8 +200,16 @@ func (rs *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 	defer nodePortOp.Finish()
 
 	if service.Spec.Type == api.ServiceTypeNodePort || service.Spec.Type == api.ServiceTypeLoadBalancer {
-		if err := initNodePorts(service, nodePortOp); err != nil {
-			return nil, err
+		if utilfeature.DefaultFeatureGate.Enabled(features.ServiceLBNodePortControl) {
```

Comment:
I don't think we need the feature gate check here. The `AllocateLoadBalancerNodePorts` field will be nil-ed out in `PrepareForCreate` or `PrepareForUpdate` if the gate was disabled so we can just check for nil field here.